src/exo/LoopIR_scheduling.py

```diff
@@ -251,68 +251,39 @@ class _PartitionLoop(LoopIR_Rewrite):
     def __init__(self, proc, loop_cursor, num):
         self.stmt = loop_cursor._node()
         self.partition_by = num
-        self.second = False
-        self.second_iter = None
 
         super().__init__(proc)
 
-        self.proc = InferEffects(self.proc).result()
-
     def map_s(self, s):
         if s is self.stmt:
             assert isinstance(s, LoopIR.Seq)
 
             if not isinstance(s.hi, LoopIR.Const):
-                raise SchedulingError("expected loop bound to be constant")
+                raise SchedulingError("expected loop bound to be a literal")
 
             if s.hi.val <= self.partition_by:
                 raise SchedulingError(
                     "expected loop bound to be larger than partitioning value"
                 )
 
-            body = self.apply_stmts(s.body)
-            first_loop = s.update(
-                hi=LoopIR.Const(self.partition_by, T.int, s.srcinfo),
-                body=body,
-                eff=None,
-            )
-
-            # Should add partition_by to everything in body
-            self.second = True
+            part_by = LoopIR.Const(self.partition_by, T.int, s.srcinfo)
+            loop1 = type(s)(s.iter, part_by, s.body, None, s.srcinfo)
```

Comment:
This should use `.update()`
```suggestion
            loop1 = s.update(hi=part_by, eff=None)
```